src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -48,31 +57,34 @@
  * @author lkuehne
  */
 public class PackageObjectFactory implements ModuleFactory {
+    /** Base package of checkstyle modules checks. */
+    public static final String BASE_PACKAGE = "com.puppycrawl.tools.checkstyle";
+
+    /** Exception message when it is unable to create a class instance. */
+    public static final String UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE =
+            "PackageObjectFactory.unableToInstantiateExceptionMessage";
+
+    /** Suffix of checks. */
+    public static final String CHECK_SUFFIX = "Check";
+
     /** Map of Checkstyle module names to their fully qualified names. */
     private static final Map<String, String> NAME_TO_FULL_MODULE_NAME = new HashMap<>();
 
     /** Exception message when null class loader is given. */
     private static final String NULL_LOADER_MESSAGE = "moduleClassLoader must not be null";
 
-    /** Exception message when it is unable to create a class instance. */
-    private static final String UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE =
-        "PackageObjectFactory.unableToInstantiateExceptionMessage";
-
     /** Separator to use in strings. */
     private static final String STRING_SEPARATOR = ", ";
 
-    /** Suffix of checks. */
-    private static final String CHECK_SUFFIX = "Check";
-
-    /** Base package of checkstyle modules checks. */
-    private static final String BASE_PACKAGE = "com.puppycrawl.tools.checkstyle";
-
     /** A list of package names to prepend to class names. */
     private final Set<String> packages;
 
     /** The class loader used to load Checkstyle core and custom modules. */
     private final ClassLoader moduleClassLoader;
 
+    /** Map of third party Checkstyle module names to their fully qualified names. */
+    private Map<String, String> thirdPartyNameToFullModuleName;
```

Comment:
Map isn't static because we are relying on values from the constructor. This will cause us to always re-scan the classpath for more checks in our tests only.
Should we be concerned?